src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantTranchesFromInfoKey.java

```diff
@@ -0,0 +1,197 @@
+    protected void onStartup() {
+        /* check for successful import of libraries */
+        PythonScriptExecutor.checkPythonEnvironmentForPackage("vqsr_cnn");
+        trancheString = " --tranches " + tranches.stream().map(d -> Double.toString(d)).collect(Collectors.joining(" "));
+        try {
+            final String idxExt = ".tbi";
+            tempFile = File.createTempFile(outputVcf, "_temp.vcf.gz");
+            tempFile.deleteOnExit();
+            tempFileIdx = new File(tempFile.getAbsolutePath()+idxExt);
+            tempFileIdx.deleteOnExit();
+        } catch (IOException e) {
+            throw new GATKException("Error when creating temp files.", e);
+        }
+    }
+
+    @Override
+    protected Object doWork() {
+        final Resource pythonScriptResource = new Resource("tranches.py", VariantTranchesFromInfoKey.class);
+        final List<String> snpArguments = new ArrayList<>(Arrays.asList(
+                "--mode", "write_snp_tranches",
+                "--input_vcf", inputVcf,
+                "--train_vcf", snpTruthVcf,
+                "--score_keys", infoKey,
+                trancheString,
+                "--samples", Integer.toString(maxSites),
+                "--output_vcf", tempFile.getAbsolutePath()));
+
+        logger.info("SNP Args are:"+ Arrays.toString(snpArguments.toArray()));
+        final boolean pythonReturnCode = pythonExecutor.executeScript(
+                pythonScriptResource,
+                null,
+                snpArguments
+        );
+
+        final FeatureCodec<? extends Feature, ?> codec = FeatureManager.getCodecForFile(tempFile);
+        createAppropriateIndexInMemory(codec, tempFile, tempFileIdx);
+        final Index index = createAppropriateIndexInMemory(codec, tempFile, tempFileIdx);
+
+        try {
+            index.write(tempFileIdx);
+        } catch (final IOException e) {
+            throw new GATKException(String.format("Could not write temporary index to file: %s", tempFileIdx.getAbsolutePath()), e);
+        }
+
+        final List<String> indelArguments = new ArrayList<>(Arrays.asList(
+                "--mode", "write_indel_tranches",
+                "--input_vcf", tempFile.getAbsolutePath(),
+                "--train_vcf", indelTruthVcf,
+                "--score_keys", infoKey,
+                trancheString,
+                "--samples", Integer.toString(maxSites),
+                "--output_vcf", outputVcf));
+
+        logger.info("Did SNP filtering, now INDELs. Arguments are:"+ Arrays.toString(indelArguments.toArray()));
+        final boolean pythonReturnCode2 = pythonExecutor.executeScript(
+                pythonScriptResource,
+                null,
+                indelArguments
+        );
+        return pythonReturnCode && pythonReturnCode2;
+    }
+
+    // Stolen and adapted from IndexFeatureFile.java
+    private Index createAppropriateIndexInMemory(final FeatureCodec<? extends Feature, ?> codec, File featureFile, File indexFile) {
```

Comment:
This code has some weird failure modes, some of which would be weird, i.e., throwing a user exception telling the user it can't index some temporary file the user never heard of.

Reply:
Yes, I agree... I think the best thing is to just re-write from scratch in java.